### DISCUSSION TITLE
[V3] livewire:navigating Listener only once time work

### DISCUSSION THREAD

**Discussion by codenashwan** [livewire/livewire#6901] [1. Help]
Livewire + TinyMCE Integration Issue

I am encountering an issue with the integration of Livewire (v3) and TinyMCE in my Laravel application. When users visit the "Add Post" page, the textarea is supposed to be replaced with a TinyMCE rich text editor. However, I'm experiencing a problem where the editor works fine the first time a user navigates to the page but doesn't work when the user returns to the page for a second time.

**Environment:**
Laravel Livewire v3
TinyMCE v6
Laravel v10.10

home.blade.php : 
`    <x-button wire:navigate href="{{ route('user.posts.add') }}" blue squared icon="plus" label="Add New Post" />
`

add.blade.php :

```
<form wire:submit="save" class="flex flex-col gap-3">
        <div class="grid grid-cols-1 lg:grid-cols-3 gap-4">
            <x-input required wire:model="title" placeholder="Title Post" label="Title Post" />
            <x-input required wire:model="photo" type="file" label="Photo Of Post" />
        </div>
        <div wire:ignore>
            <x-textarea required wire:model="content" placeholder="Content" id="tiny" rows="9" />
        </div>
        <div class="grid grid-cols-1 lg:grid-cols-3 gap-4">
            <x-select label="Type Post" placeholder="Type Post" wire:model="category_id" option-label="name"
                option-value="id" :options="$categories" />
        </div>
        <x-errors />
        <div class="flex mt-3">
            <x-button type="submit" squared blue lg label="Add New Post" spinner="save" />
        </div>
    </form>
    
```
    
    
    rescourse/app.js
    
    
   ```
 document.addEventListener("livewire:navigating", () => {
    tinymce.init({
        setup: function (editor) {
            editor.on("init change", function () {
                editor.save();
            });
            editor.on("blur", function (e) {
                Livewire.dispatch("setContent", {
                    html: editor.getContent(),
                });
            });
        },
        selector: "#tiny",
        plugins:
            "advlist preview  help code pagebreak anchor autolink charmap codesample emoticons image link lists media searchreplace table visualblocks wordcount",
        toolbar:
            "undo redo | blocks fontfamily fontsize | bold italic underline strikethrough | link image media table | align lineheight | numlist bullist indent outdent | emoticons charmap | removeformat",
        toolbar_mode: "floating",
        extended_valid_elements:
            "iframe[src|width|height|name|align|frameborder|scrolling|marginheight|marginwidth|allowfullscreen|class|preload|controls|autoplay|loop|poster]",
    });
});
```

https://github.com/livewire/livewire/assets/35005761/17d92062-e70a-4e61-a7a3-83687a4eed02

**Comment by N1ebieski** [comment]
Why do you init a tinymce in the ```livewire:navigating``` listener instead in the alpine component?

I implemented a tinymce in my tiny livewire starter, so you can check repo, maybe it will help you:

1. [Tinymce alpine component](https://github.com/N1ebieski/livewire-starter/blob/main/resources/js/tinymce/tinymce.js)
2. [Tinymce blade component](https://github.com/N1ebieski/livewire-starter/blob/main/resources/views/components/forms/tinymce-component.blade.php)
